src/Toml/Parser.hs

```diff
@@ -159,10 +161,19 @@ tableNameP = lexeme $ between (char '[') (char ']') keyP
 
 -- Values
 
+decimalP :: Parser Integer
+decimalP = mkInteger <$> decimalStringP
+  where
+    decimalStringP   = fst <$> match (some digitChar >> many _digitsP)
+    _digitsP         = try (char '_' >> some digitChar)
+    mkInteger        = textToInt . stripUnderscores
+    textToInt        = fst . fromRight undefined . TR.decimal
```

Comment:
`fromRight undefined` is not a pleasant thing... But usually instead of `undefined` it's always better to write `error` with some description, like:

```haskell
fromRight (error "Underscore parser has a bug")
```